### DISCUSSION TITLE
How to process text before copying using mouse click?

### DISCUSSION THREAD

**Discussion by siddhpant** [kovidgoyal/kitty#7699] [Q&A] [answered]
I have ctrl+right enabled for copying to clipboard.

But for git hashes, I only want to copy the first 12 characters even if I selected the entire thing. Basically, `selection[:12]`. I want to map this stripping flow to ctrl+triplepress right.

Is there a way to achieve this?

**Comment by kovidgoyal** [comment]
Use the hints kitten, it can do arbitrary things to pices of text on the
screen. You will trigger it with keyboard but once triggerred you can
click on matches to select them.

**Comment by siddhpant** [comment, marked as answer]
I achieved this with the following kitten:

```
from kitty.boss import Boss
from kitty.clipboard import set_clipboard_string
from kittens.tui.handler import result_handler


def main(args: list[str]) -> None:
    """
    See "Using kittens to script kitty, without any terminal UI" section of
    https://sw.kovidgoyal.net/kitty/kittens/custom
    """
    pass
# End of main().


@result_handler(no_ui=True)
def handle_result(
    args: list[str],
    _,
    target_window_id: int,
    boss: Boss
) -> None:
    """
    Custom kitten to copy first 12 chars of the selection.
    To be used for copying git hashes in short form quickly.
    """
    window = boss.window_id_map.get(target_window_id)
    if window is None:
        return

    selection = boss.call_remote_control(window, ("get-text", "--self",
                                                  "--extent=selection"))

    set_clipboard_string(selection[:12])
# End of handle_result().


# End of file.
```

And then mapped it using `mouse_map ctrl+alt+right press ungrabbed kitten <filename>`.

I didn't use triplepress since it also includes the single press which I have mapped to copy, so initially the entire text gets copied which I didn't want to do.
